Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+    "groundwater outflow",
+    "groundwater head",
+    "moisture content",
+    "concentration of pollutant"
+};
+
+const std::array<std::string,7> node_vars =
+{
+    "water depth",
+    "hydraulic head",
+    "volume of stored water",
+    "lateral inflow",
+    "total inflow",
+    "flow lost to flooding",
+    "concentration of pollutant"
+};
+
+const std::array<std::string,6> link_vars =
+{
+    "flow rate",
+    "flow depth",
+    "flow velocity",
+    "flow volume",
+    "fraction conduit/non-conduit",
+    "concentration of pollutant"
+};
+
+const std::array<std::string,15> system_vars =
+{
+    "air temperature",
+    "rainfall",
+    "snow depth",
+    "evaporation + infiltration loss",
+    "runoff flow",
+    "dry weather inflow",
+    "groundwater inflow",
+    "RDII inflow",
+    "direct inflow",
+    "total lateral inflow",
+    "flow lost to flooding",
+    "flow leaving through outfalls",
+    "volume of stored water",
+    "actual evaporation rate",
+    "potential evaporation rate"
+};
+
+std::array<std::size_t,4> const n_obj_params = { 8, 6, 5, 15 };
+
+SwmmInterface* SwmmInterface::create(std::string const& file_name)
+{
+    if (file_name.length() < 5)
+        return nullptr;
+
+    if (!(SwmmInterface::isSwmmInputFile(file_name) || SwmmInterface::isSwmmOutputFile(file_name)))
+        return nullptr;
+
+    std::string const base_name (file_name.substr(0, file_name.length() - 4));
+    SwmmInterface* swmm = new SwmmInterface(base_name);
+    if (swmm->convertSwmmInputToLineMesh())
+        return swmm;
+
+    ERR ("Error creating mesh from SWMM file.");
+    delete swmm;
+    return nullptr;
+}
+
+SwmmInterface::SwmmInterface(std::string const& swmm_base_name)
+: _base_name (swmm_base_name), _mesh(nullptr)
+{
+}
+
+SwmmInterface::~SwmmInterface()
+{
+    for (Subcatchment sc  : _subcatchments)
+        delete sc.outline;
+
+    for (GeoLib::Point* pnt : _subcatchment_points)
+        delete pnt;
+}
+
+bool SwmmInterface::isSwmmInputFile(std::string const& inp_file_name)
+{
+    std::string path (inp_file_name);
+    std::transform(path.begin(), path.end(), path.begin(), tolower);
+    if (BaseLib::getFileExtension(path) != "inp")
+    {
+        ERR ("SWMMInterface: file extension %s not recognised (should be *.inp).",
+            BaseLib::getFileExtension(inp_file_name).c_str());
+        return false;
+    }
+
+    std::ifstream in ( inp_file_name.c_str() );
+    if (!in.is_open())
+    {
+        ERR ("SWMMInterface: Could not open input file %s.", inp_file_name.c_str());
+        return false;
+    }
+
+    std::string line;
+    bool header_found (false);
+    std::size_t pos_beg (0), pos_end (0);
+    while (!header_found)
+    {
+        getline(in, line);
+        pos_beg = line.find_first_not_of(' ', pos_end);
+        pos_end = line.find_first_of(" \n", pos_beg);
+
+        if (line.empty() || pos_beg==pos_end || line.compare(pos_beg,2,";;") == 0)
+            continue;
+
+        if (line == "[TITLE]")
```

Comment:
It's OK, getline docu says '\n' is not included in the read string, as far as I remember.